server/test/src/scala/lsug/ServerSpec.scala

```diff
@@ -0,0 +1,86 @@
+package scala.lsug
+
+import cats.effect.{Blocker, IO}
+import lsug.protocol.Meetup.MeetupDotCom
+import lsug.protocol.Meetup.MeetupDotCom.Event
+import lsug.{IOSuite, Meetup, Server, protocol}
+
+import java.nio.file.Paths
+import java.time.{LocalDate, LocalDateTime, LocalTime, ZoneId, ZonedDateTime}
+
+final class ServerSpec extends IOSuite {
+
+  class MockedMeetup extends Meetup[IO] {
+    override def event(id: Event.Id): IO[Option[MeetupDotCom.Event]] = IO(None)
+  }
+  val mockedMeetup = new MockedMeetup
+
+  test("eventsBefore return a list of past events ordered by time") {
+
+    Blocker[IO].use { blocker =>
+      val server: Server[IO] = new Server[IO](
+        Paths.get("./server/src/main/resources").toAbsolutePath,
+        mockedMeetup,
+        blocker
+      )
+      val result = server.eventsBefore(
```

Comment:
Testing with time is tricky, as you want your tests to pass regardless of the time and timezone you run them at.
When writing time related tests, it's worth asking yourself:
 - would this fail if I ran it in the summer?
 - would this fail if I ran it in a different timezone? (what if you ran it in San Francisco, for example?)

I think this test would fail if we ran it in Paris in the summer just after an event has finished. This is because London switches from being UTC to BST, but Paris is in UTC.

Let's say it's currently 1PM UTC (2PM BST), and an event ended half an hour ago at 12:30PM UTC (1:30PM BST).
 - We ask for all events before 1PM UTC.
 - The server returns the event that just ended, but in BST time.
 - We compare the event time (1:30PM) with the current time (1PM), and find that the event time was later.

You can fix this by comparing the event times with the local time in London:

```scala
val timeUTC = LocalDateTime.now().atZone(ZoneId.of("UTC"))
val localTimeLondon = timeUTC.withZoneSameInstant(ZoneId.of("Europe/London")).toLocalDateTime
val result = server.eventsBefore(timeUTC)
  ...
    eventTimes.foldLeft(localTimeLondon)(...)
```

Note that `LocalDateTime.now().atZone(ZoneId.of("UTC"))` isn't the current time in UTC. If you ran this at 1PM in San Francisco (GMT - 8), the current time would be 9PM UTC, but the `timeUTC` would be 1PM UTC. 
Can you also recode this such that `timeUTC` is actually the current time? It doesn't affect whether the test passes, but makes it a bit easier to understand.

Reply:
I changed "UTC" to "Europe/London". Is that right?

Reply:
That works, but isn't actually a time in UTC (it's a time in London, which is the same for half the year). I've reworked it to a UTC `ZonedDateTime` for clarity.